### PR TITLE
Automatically add `from __future__ import annotations`

### PR DESCRIPTION
**Pull request by kaxil**
This PR automatically adds `from __future__ import annotations` to all the Python files so that we can take advantage of the lazy imports

It uses https://pycqa.github.io/isort/docs/configuration/options.html#add-imports from isort

### CODE REVIEW

File: python-sdk/example_dags/example_dataframe_api.py [hunk 1]
```diff
@@ -41,7 +42,7 @@ def load_and_group_covid_data():
 
 
 @aql.dataframe(columns_names_capitalization="original")
-def find_worst_covid_month(dfs: List[pd.DataFrame]):
+def find_worst_covid_month(dfs: list[pd.DataFrame]):
```
Comment (pankajkoti) [inline python-sdk/example_dags/example_dataframe_api.py:45]: Did you have to make such changes explicitly or we were able to do this automatically?

**Review by pankajkoti — APPROVED** [review]
@feluelle @tatiana SQL CLI tests are failing with the below error: 
```FAILED tests/test_run_dag.py::test_run_dag_with_conn_id - AttributeError: 'Table' object has no attribute 'iloc'```
 Any idea?

File: python-sdk/example_dags/example_dataframe_api.py (hunk 1 above)
Comment (pankajkoti) [reply python-sdk/example_dags/example_dataframe_api.py:45]: Looks like the test is failing only on this PR. It seems stable on other PRs and on main.

Comment (kaxil) [reply python-sdk/example_dags/example_dataframe_api.py:45]: I have rebased the PR, let's see :)

**Comment by feluelle** [comment]
I don't get it - why is that failing?!

I can debug this tomorrow, because I think it is really useful to have.

**Comment by feluelle** [comment]
Simply removing the `from __future__ import annotations` only from the sqlite_test.py DAG file fixes the issue. Continuing to debug the issue..

**Comment by feluelle** [comment]
@kaxil Okay, I found out what the issues is. We basically are checking for the class type in [here](https://github.com/astronomer/astro-sdk/blob/main/python-sdk/src/astro/sql/operators/dataframe.py#L60). So we need the imported module at this time. If we use future annotations i.e. do a lazy import the types will be translated into `'pandas.Dataframe'`, but we need it as class type. So we either import it there(if not already imported) or we cannot use lazy importing.

**Comment by tatiana** [comment]
@kaxil I have mixed feelings about this PR!
What are we gaining by adding this to all files? How often do we need lazy import? Wouldn't it be better to be more explicit about it - if and when we need it?

**Comment by kaxil** [comment]
>What are we gaining by adding this to all files? How often do we need lazy import? Wouldn't it be better to be more explicit about it - if and when we need it?

There is absolutely no reason to not have lazy imports. We should not need to do if TYPE_CHECKING, it was the limitation of python earlier, but the devs are addressing the same issue.

I will see where I can get to, currently there are conflicts on the PR and tests failing

**Comment by feluelle** [comment]
Yes, if it works it is a nice feature, but as you already noticed currently for some features we rely on direct imports.

**Comment by tatiana** [comment]
Since the SQL CLI was discontinued, I'm closing this PR.

**Pull request closed without merging**